drivers/volume/portworx/pure.go

```diff
@@ -18,3 +25,21 @@ func (p *pure) Init(sched, nodeDriver, token, storageProvisioner, csiGenericDriv
 func (p *pure) String() string {
 	return PureDriverName
 }
+
+func (p *pure) ValidateCreateSnapshot(volumeName string, params map[string]string) error {
+	var token string
+	token = p.getTokenForVolume(volumeName, params)
+	if val, hasKey := params[refreshEndpointParam]; hasKey {
+		refreshEndpoint, _ := strconv.ParseBool(val)
+		p.refreshEndpoint = refreshEndpoint
+	}
+
+	volDriver := p.getVolDriver()
+	// This is the only difference: we have to name snapshots with hyphens, not underscores
+	_, err := volDriver.SnapshotCreate(p.getContextWithToken(context.Background(), token), &api.SdkVolumeSnapshotCreateRequest{VolumeId: volumeName, Name: volumeName + "-snapshot"})
```

Comment:
Why do we need to create the snapshot when validating a snapshot?

Reply:
I was kinda wondering that originally myself too. I think it's "Validating creating a snapshot". e.g. validating that the creation is successful?

Reply:
@david-enli, do you have the context on this?

Reply:
Discussed offline in Slack, this has been resolved. Added comments for notes to refactor later